### PR TITLE
[LibraryImportGenerator] Update diagnostic titles to use localized strings

### PR DESCRIPTION
**Pull request by elinor-fung**
The analyzer diagnostics were using localized strings for the message and description, but not the title.

Also collapses a few of the more specific IDs into existing ones.

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]
Look at someone updating all the xlf files :-P

Do we need to update the shipped diagnostics in the txt file? I thought they were the numbered values.

**Comment by elinor-fung** [comment]
Are you talking about the [list-of-diagnostics](https://github.com/dotnet/runtime/blob/main/docs/project/list-of-diagnostics.md) file? We will. We still need to switch these all to use the `SYSLIB` prefix (planning on doing that after this change) and will update that file at the same time.

File: src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomTypeMarshallerAnalyzer.cs [hunk 1]
```diff
@@ -345,7 +345,7 @@ public static class MissingMemberNames
         public static readonly DiagnosticDescriptor LinearCollectionElementTypesMustMatchRule =
             new DiagnosticDescriptor(
                 Ids.InvalidSignaturesInMarshallerShape,
-                "InvalidSignaturesInMarshallerShape",
```
Comment (danmoseley) [inline src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomTypeMarshallerAnalyzer.cs:348]: Just curious do we consistently use Marshaller spelling? As opposed to Marshaler.

Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomTypeMarshallerAnalyzer.cs:348]: As of two-ish years ago, yes. Our new APIs / code should be consistently using the double `l`. We settled on: marshal, marshalling, marshalled, marshaller. We've also been working with docs style guide owners to try to get an entry added.

**Comment by elinor-fung** [comment]
Failure is https://github.com/dotnet/runtime/issues/66571

**Pull request merged**